### ISSUE TITLE
What should happen when peers try to sync up, but their Automerge docs are completely divergent?

### ISSUE THREAD

**Issue by blinry** [ethersync/ethersync#69]
This could happen, for example, if both sides `--init`.

They should probably prevented from connecting?

**Comment by zormit** [comment]
We should also think about some kind of versioning of the persisted data format. Whenever we change the structs of our CRDT something could potentially break when deserializing that data.

I just hopped between versions and this happened at startup of a peer (which tried to sync to another peer that had some old/outdated serialized data-format, _I assume_):

```
13:42:55  WARN ThreadId(11) ethersync::types: Resolved conflict for file 'file3' by overwriting your version
thread 'tokio-runtime-worker' panicked at src/types.rs:118:21:
Failed to convert patch to delta: Error("missing field `userid`", line: 1, column: 92)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
mn@pom playground2 on branch tags/v0.3.0
```

**Comment by blinry** [comment]
Note: The versioning also seems less urgent once we can go back to top-level filenames, as explained [here](https://github.com/ethersync/ethersync/pull/152#issuecomment-2311619197). But once we go more "stable", thinking about versioning (also in the editor protocol) seems like a good thing!

**Comment by blinry** [comment]
In any case, the error message if this happens should be much better.

**Comment by zormit** [comment]
Yeah error message is helpful indeed. I think I love the solution number 4. Maybe I am biased because I came to the same conclusion the documentation suggests. It's a bit like a DB schema. I wonder if it can be part of the _build_ process in some way, s.t. we don't need to put the binary into the repo. But then it might be tricky to get the same hashes between two builds 🤔 

But it seems, at least in their example, that this byte array can live in the code... So it's not a binary blob, but "just a bit more" than a hardcoded string or so?

I did a small test:
```diff
diff --git a/daemon/src/document.rs b/daemon/src/document.rs
index caa69e3..73a7841 100644
--- a/daemon/src/document.rs
+++ b/daemon/src/document.rs
@@ -109,6 +109,7 @@ impl Document {
         self.doc
             .put_object(automerge::ROOT, "states", ObjType::Map)
             .expect("Failed to initialize states Map object");
+        debug!("initialized {:?}", self.doc.save());
     }

     pub fn initialize_text(&mut self, text: &str, file_path: &str) {

```

gave me

```
3:48:18 DEBUG ThreadId(01) ethersync::document: initialized [133, 111, 74, 131, 38, 199, 162, 39, 0, 118, 1, 16, 143, 191, 175, 25, 231, 200, 78, 169, 178, 203, 13, 157, 59, 180, 95, 143, 1, 106, 24, 124, 208, 68, 65, 209, 161, 206, 156, 38, 30, 89, 41, 75, 109, 164, 70, 196, 175, 147, 250, 142, 101, 21, 99, 68, 154, 72, 28, 227, 151, 6, 1, 2, 3, 2, 19, 2, 35, 2, 64, 2, 86, 2, 7, 21, 14, 33, 2, 35, 2, 52, 1, 66, 2, 86, 2, 128, 1, 2, 127, 0, 127, 1, 127, 2, 127, 0, 127, 0, 127, 7, 126, 5, 102, 105, 108, 101, 115, 6, 115, 116, 97, 116, 101, 115, 2, 0, 2, 1, 2, 2, 0, 2, 0, 2, 0, 0]
```
which are 128 bytes.

**Comment by blinry** [comment]
By now, we picked solution #4 above, and implemented it in #214.

If both sides would have peers configured, and then use `ethersync share --init`, they would get a random mix of their file versions (if both sides have files of the same names). I think that's a bit surprising, but better than a crash.

I'd suggest we close this issue for now, until we have a more concrete problem. The outcome of #97 might solve this, anyway.
